packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -395,6 +416,374 @@ class FocusScopeNode extends Object with DiagnosticableTreeMixin {
   }
 }
 
+typedef FocusableNodeVisitor = bool Function(FocusableNode node);
+
+/// An interface for data from policy objects to be held by the scope nodes.
+///
+/// This data is set and requested via [FocusableNode.policyData].
+///
+/// If a policy data object encounters a type of policy data that it doesn't
+/// recognize, then it should not use it, but may overwrite it (this may happen
+/// when the traversal policy is changed).
+abstract class PolicyData {
+  /// A const constructor so that subclasses can be const.
+  const PolicyData();
+}
+
+/// A node used by [Focusable] to represent and change the input focus.
```

Comment:
It's not just used by Focusable, right?

I think a layering approach to this would be to define this in a standalone way, and then to present Focusable as a convenience widget that wraps this object, rather than presenting this as an implementation detail of Focusable.